tools/circt-test/circt-test.cpp

```diff
@@ -240,6 +246,21 @@ static StringRef toString(TestKind kind) {
   return "unknown";
 }
 
+static void collectRunnerFilters(DictionaryAttr attrs, StringRef attrName,
+                                 llvm::SmallPtrSetImpl<StringAttr> &names) {
+  auto attr = attrs.get(attrName);
+  if (!attr)
+    return;
+  if (auto arrayAttr = dyn_cast<ArrayAttr>(attr)) {
+    for (auto elementAttr : arrayAttr.getValue())
+      if (auto stringAttr = dyn_cast<StringAttr>(elementAttr))
+        names.insert(stringAttr);
+    return;
+  }
+  if (auto stringAttr = dyn_cast<StringAttr>(attr))
+    names.insert(stringAttr);
```

Comment:
Allowing null/array/string looks a bit too untyped to me. Can we at least reject StringAttr case? Also it would be nice to have a document for parameters.

Reply:
Yeah that's a good point. I'll make this fallible and add some error reporting. I can add the attributes that have some specific meaning to the op definition, and then later on we can maybe move them to the docs where they're easier for users to find?

Reply:
Sounds great, thanks!